packages/rocketchat-api/server/v1/settings.js

```diff
@@ -1,6 +1,34 @@
 import _ from 'underscore';
 
 // settings endpoints
+RocketChat.API.v1.addRoute('settings.public', { authRequired: false }, {
+	get() {
+		const { offset, count } = this.getPaginationItems();
+		const { sort, fields, query } = this.parseJsonQuery();
+
+		let ourQuery = {
+			hidden: { $ne: true },
+			'public': true
+		};
+
+		ourQuery = Object.assign({}, query, ourQuery);
+
+		const settings = RocketChat.models.Settings.find(ourQuery, {
```

Comment:
Should this be changed to a model call? We already have `RocketChat.models.Settings.findNotHiddenPublic` which has the same filter, just need to implement additional `sort`, `skip` and `limit` options.

I also think this should only return `_id` and `value` fields, I cannot see any use for the other fields.

Reply:
The ideia is to have an API that is versatile, that's why the API pass the query and allow the caller to pass more filters and require other fields if necessary

Reply:
my only concern is how it can be abused.

but still bypassing the model (using .`find` directly) defeats model's purpose and I thought was strongly discouraged.

Reply:
I'm going to agree that this particular endpoint should use the `findNotHIddenPublic` with pagination support, however adding that support requires additional work and can be done via a secondary pull request after this one is merged. This way we don't delay any longer getting this out and come back to it.